traits/trait_base.py

```diff
@@ -314,10 +314,13 @@ def get_resource_path ( level = 2 ):
     # open/read commands, the app bundlers will look for resources (i.e.  data
     # files, images, etc.) in specific locations.  For py2app, this is in the
     # [myapp].app/Contents/Resources directory.  For py2exe, this is the same
-    # directory as the [myapp].exe executable file generated by py2exe.
+    # directory as the [myapp].exe executable file generated by py2exe.  For
+    # pyinstaller, the attribute sys._MEIPASS is set to this directory.
     frozen = getattr(sys, 'frozen', False)
     if frozen:
-        if frozen == 'macosx_app':
+        if hasattr(sys, "_MEIPASS"):
+            root = sys._MEIPASS
+        elif frozen in ('macosx_app'):
```

Comment:
This doesn't look right: either `('macosx_app')` should be replaced with a length-one tuple, or the line should be restored to its previous `==` form. Otherwise we're just checking whether `frozen` is one of the letters `'m'`, `'a'`, `'c'`, etc.